### ISSUE TITLE
Add small/synthetic tests

### ISSUE THREAD

**Issue by swfsql** [swfsql/burn-mamba#5]
Burn has an [modern-lstm example](https://github.com/tracel-ai/burn/blob/9c175cfdc08b6f5ae199df765e79e25928827a80/examples/modern-lstm/README.md) that is useful for both verifying if the model can be trained, and how to effectively use it. This kind of things should be useful in here.

- [x] Avoid deleting files: Currently the examples create directories under `/tmp`, but they delete previously generated files. It is safer to avoid deletion (never use them) and create new directories for repeated runs. Perhaps a os-generic tmp folder selection could also be used.
  - This was improved on https://github.com/swfsql/burn-mamba/commit/277d2d155f1377cdda0c40534bbcd50806e7a8db.

Some tests or examples:
- [x] Add Fibonacci-like test ([modern-lstm example](https://github.com/tracel-ai/burn/blob/9c175cfdc08b6f5ae199df765e79e25928827a80/examples/modern-lstm/README.md)) -- input: 1, output: 1, sequence: 10.
- [x] [MNIST digit classification](https://github.com/tracel-ai/burn/blob/60b8d227b1f6555fd483271b92815ca0bc9203ae/examples/mnist/README.md) --  input: 1, output: 10, sequence: 28*28.
  - Read a sequence pixels and read the last timestep output as the predicted classification.
  - [ ] Extra: Add a bidi-variant example (which hopefully may reach higher acc).
- [ ] Distillation?
  - Have a teacher model (e.g. from mnist classification) teach a student one. Test if whitenoise distillation with soft-labels (logits or post-softmax classification) targetting works.
- [ ] MNIST auto-encoder -- input: 1, output: 1, sequence: 28*28; or a patched variant.
  - Mamba-2 works fine for encoding (extracting features), not so much for decoding (data generation).  
  The naive way of using it is to have a single latent and expand it into a sequence, but the resulting constant input is quite limiting for a mamba layer.
  Alternatively, if parallel training (for the decoder) is discarded, it is possible to do step-wise generation and pass the last sample as part of the input for the next timestep, but in this case it may be preferable to combine the Mamba-2 block to a diffusion architecture, which hopefully preserves parallel/ssd processing.
  - Extra: Once the decoder is figured out, test various types of latent-encoders:
    - [ ] Simple AE with latent as a linear projection from the hidden state. Possibly non-causal for the encoder.
    - [ ] SoftVQ-VAE as an extension from the simple AE. Possibly non-causal for the encoder.
- [ ] Text prediction.
  - Perhaps base on karpathy's Shakespeare texts, or use the TinyStories dataset (where sub-10M param models should be able to give interesting results).
  - Perhaps allow for different tokenizers to be used. Or even try training an auto-encoder based on Mamba itself, for tokenizing/encoding.
- [ ] More examples?
- [ ] Extra: Add a trained model for each kind of example, so users may just test by inference instead of training their own.
